contracts/interchain-token-service/src/contract.rs

```diff
@@ -238,16 +240,39 @@ impl InterchainTokenServiceInterface for InterchainTokenService {
     }
 
     fn deploy_remote_interchain_token(
-        _env: &Env,
-        _caller: Address,
-        _salt: BytesN<32>,
-        _minter: Option<Bytes>,
-        _destination_chain: String,
-        _gas_token: Token,
+        env: &Env,
+        caller: Address,
+        salt: BytesN<32>,
+        minter: Option<Address>,
+        destination_chain: String,
+        gas_token: Token,
     ) -> Result<BytesN<32>, ContractError> {
-        // TODO: implementation
+        caller.require_auth();
+
+        let deploy_salt = Self::interchain_token_deploy_salt(env, caller.clone(), salt);
+        let token_id = Self::interchain_token_id(env, Address::zero(env), deploy_salt);
+
+        if let Some(ref minter) = minter {
+            ensure!(
+                Self::check_token_minter(env, token_id.clone(), minter.clone()),
+                ContractError::InvalidMinter
+            );
+        }
+
+        let registered_token_address = Self::token_id_config(env, token_id.clone()).token_address;
```

Comment:
```suggestion
        let registered_token_address = Self::token_id_config(env, token_id.clone()).token_address;
```
Use the token_address query instead?